Sources/CacheContainer.swift

```diff
@@ -1,23 +1,23 @@
 import Foundation
 
-public final class CacheContainer {
+public final class CacheContainer: Sendable {
     static let shared = CacheContainer()
 
     public static func clearAll() {
-        shared.storage.forEach { (key, _) in
-            shared.storage.removeValue(forKey: key)
+        shared.storage.keys.forEach {
+            shared.storage.removeValue(forKey: $0)
         }
     }
 
     public static func clearAll(where shouldBeCleared: (CacheKey) -> Bool) {
-        shared.storage.forEach { (key, _) in
-            if shouldBeCleared(key) {
-                shared.storage.removeValue(forKey: key)
+        shared.storage.keys.forEach {
+            if shouldBeCleared($0) {
+                shared.storage.removeValue(forKey: $0)
             }
         }
     }
 
-    var storage: [CacheKey : Data] = [:]
+    let storage: ThreadSafeDictionary<CacheKey, Data> = .init()
```

Comment:
I wonder if the crash issue cannot be resolved simply by wrapping `storage` with `OSAllocatedUnfairLock`.

Reply:
Thanks, I'll try it.

Reply:
@ichiho-ojima 
`OSAllocatedUnfairLock` requires iOS 16+, tvOS 16+, watchOS 9+, and macOS 13+. Can we bump our minimum deployment targets to these versions?

Reply:
Oops, I forgot to consider the version where `OSAllocatedUnfairLock` is available.
Personally, I think it's an option to make the supported version iOS 16+ for using `OSAllocatedUnfairLock`, but what do you think? @elmetal

Reply:
@ichiho-ojima 
I think that's a good approach, but we need to be mindful of versioning.